### PR TITLE
WIP - linting and ci optimizations

### PR DESCRIPTION
**Pull request by cloudify**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,11 +1,13 @@
+defaults: &defaults
+  working_directory: ~/italia-app
+  docker:
+    - image: circleci/node:8
```
Comment (fpersico) [inline .circleci/config.yml]: Why we are going to include the image here?  It is only usefull for js-related jobs. Isn't better to have a root default with only the really global properties and than a separate default_js, that inherits from default and add the node image. The, atm only 2, js-related jobs will use default_js.

Comment (cloudify) [reply .circleci/config.yml]: I thought it wasn't a big deal having just one job not using the default image, you objection makes sense though, we can have a `defaults_js` that inherits from `defaults` and use that for the js jobs:

``` yaml
defaults: &defaults
  working_directory: ~/italia-app

defaults_js: &defaults_js
  <<: *defaults
  docker:
    - image: circleci/node:8
```

**Comment by cloudify** [comment]
@fpersico looks like we're good with this, what do you think?

**Comment by fpersico** [comment]
Ok for me. We can merge.

**Pull request merged**